### ISSUE TITLE
[BUG] Incorrect Gradient Flow in Alpha-Loss Calculation of SAC Implementation

### ISSUE THREAD

**Issue by songyuc** [haosulab/ManiSkill#899]
Hi,
We've identified an implementation discrepancy in the SAC algorithm's alpha loss calculation that causes unintended gradient propagation through the policy network.

**Affected Code**  
`examples/baselines/sac/sac.py` line 423:
```python
alpha_loss = (-log_alpha.exp() * (log_pi + target_entropy)).mean()
```

**Technical Analysis**  
1. **Theoretical Violation**: According to SAC's original formulation (Haarnoja et al. 2018), the entropy term should be treated as a constant when optimizing the temperature parameter α. The current implementation allows gradients to backpropagate through the policy network via `log_pi`.
2. **Consequences**:
   - Creates unintended coupling between α optimization and policy network parameters
   - May lead to monotonically decreasing α values due to unidirectional gradient flow
   - Reduces algorithm's ability to maintain target entropy
3. **Empirical Evidence**: All reference implementations (Spinning Up, CleanRL, etc.) detach `log_pi` when calculating α loss.

**Suggested Fix**  
```python
alpha_loss = (-log_alpha.exp() * (log_pi.detach() + target_entropy)).mean()
```

**Validation Method**  
```python
# Check gradient propagation
alpha_loss.backward() 
print(actor.fc_mean.weight.grad)  # Should be None if properly detached
```

**References**  
1. SAC Paper Eq(20): [arXiv:1812.05905](https://arxiv.org/abs/1812.05905)  
2. Spinning Up Implementation: [sac.py#L242-L243](https://github.com/openai/spinningup/blob/master/spinup/algos/pytorch/sac/sac.py#L242-L243)  
3. CleanRL Implementation: [sac_continuous_action.py#L303](https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/sac_continuous_action.py#L303)

**Comment by StoneT2000** [comment]
This is not needed. The code is the same since we use torch no grad. And the referenced cleanrl code is not related at all either.
